README.rst

```diff
@@ -42,16 +63,137 @@ Add the AutoExtract downloader middleware in the settings file::
+            # - another through providers to build the article argument
+            yield article.to_item()
+
+Note that on the example above, we're going to perform two requests:
+
+* one goes through Scrapy (it might use Crawlera, Splash or no proxy at all, depending on your configuration)
+* another goes through AutoExtract API using `scrapinghub-autoextract`_
+
+If you don't need the additional request going through Scrapy,
+you can annotate the response argument of your callback with ``DummyResponse``.
+This will ignore the Scrapy request and only the AutoExtract API will be fetched.
+
+For example::
+
+    import scrapy
+    from autoextract_poet.page_inputs import AutoExtractArticleData
+    from scrapy_poet.utils import DummyResponse
+
+    class SampleSpider(scrapy.Spider):
+
+        name = "sample"
+
+        def parse(self, response: DummyResponse, article: AutoExtractArticleData):
+            # We're making a single request here to build the article argument
+            yield article.to_item()
+
+Configuration
+^^^^^^^^^^^^^
+
+First, you need to configure scrapy-poet as described on `scrapy-poet's documentation`_
+and then enable AutoExtract providers by putting the following code to Scrapy's ``settings.py`` file::
+
+    # Install AutoExtract providers
+    import scrapy_autoextract.providers
+    scrapy_autoextract.providers.install()
+
+    # Enable scrapy-poet's provider injection middleware
+    DOWNLOADER_MIDDLEWARES = {
+        'scrapy_poet.InjectionMiddleware': 543,
+    }
+
+    # Configure Twisted's reactor for asyncio support on Scrapy
+    TWISTED_REACTOR = 'twisted.internet.asyncioreactor.AsyncioSelectorReactor'
+
+Currently, our providers are implemented using asyncio.
+Scrapy has introduced asyncio support since version 2.0
+but as of Scrapy 2.3 you need to manually enable it by configuring Twisted's default reactor.
+Check `Scrapy's asyncio documentation`_ for more information.
+
+Checklist:
+
+* scrapy-poet is installed and downloader/injector middleware is configured
+* autoextract-poet is installed (page inputs are imported from this lib)
+* providers are installed on settings.py
+* Scrapy's asyncio support is enabled on settings.py
+
+Now you should be ready to use our AutoExtract providers.
+
+Exceptions
+^^^^^^^^^^
+
+While trying to fetch AutoExtract API, providers might raise some exceptions.
+Those exceptions might come from scrapy-autoextract providers themselves,
+`scrapinghub-autoextract`_, or Tenacity, the library used to implement retries.
+For example:
+
+* ``scrapy_poet.providers.QueryError``: raised when a `Query-level error`_ is returned even after retrying the request
+* ``autoextract.aio.errors.RequestError``: raised when a `Request-level error`_ is returned
+* ``autoextract.aio.errors.QueryRetryError``: raised when it's not possible to retry a `Query-level error`_
+* ``tenacity.RetryError``: raised when it's not possible to retry an error
```

Comment:
IMO we should change the retrial code adding the parameter `reraise=True` to avoid raising these exceptions at all. See https://tenacity.readthedocs.io/en/latest/#error-handling. 

Probably better to do so in a different PR

Reply:
Would you mind creating a new issue for that?

Reply:
Sure. Created here: https://scrapinghub.atlassian.net/browse/TRIN-103

Reply:
It seems the issue was already there: https://github.com/scrapinghub/scrapinghub-autoextract/issues/11

I'll update it.